### PR TITLE
Enable reading dicom file list from stdin.

### PR DESCRIPTION
**Pull request by BlakeGFitch**
Enable reading a dicom file list from stdin to allow filtering, e.g.:

> find /foo | grep "IMA$" | dicomsort.py "" foosorted/

Update README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,12 +34,15 @@ dicomsort [options...] sourceDir targetDir/<patterns>
...
     [-k,--keepGoing] - report but ignore dupicate target files
     [-v,--verbose] - print diagnostics while processing
+    [-s,--symlink] - create a symlink to dicom files in sourceDir instead of copying them
     [-t,--test] - run the built in self test (requires internet)
     [-u,--unsafe] - do not replace unsafe characters with '_' in the path
     [--help] - print this message
 
- <patterns...> is a string defining the output file and directory
-names based on the dicom tags in the file.
+ where sourceDir is directory to be scanned or "" (null string) to read file list from stdin
```
Comment (pieper) [inline README.md]: I think it would be more standard to just omit the argument if you want it to be optional.  Then instead of checking for "" you would check against None, which would be the default value if the argument is not specified.

However in this case since leaving off the argument by mistake would just leave the program hanging waiting for data on stdin, I think it would be better to introduce and explicit flag like --fileList which would take a filename for the list of files to process or a "-" character to indicate that the list should come from stdin.

File: README.md [hunk 2]
```diff
@@ -34,12 +34,15 @@ dicomsort [options...] sourceDir targetDir/<patterns>
     [-f,--forceDelete] - remove source without confirmation
     [-k,--keepGoing] - report but ignore dupicate target files
     [-v,--verbose] - print diagnostics while processing
+    [-s,--symlink] - create a symlink to dicom files in sourceDir instead of copying them
```
Comment (pieper) [inline README.md]: This should technically be a different commit since it's documentation for the other PR.

File: dicomsort.py [hunk 3]
```diff
@@ -189,10 +189,16 @@ def renameFiles(self):
...
-            for file in files:
-                file = os.path.join(root,file)
-                allFiles.append(file)
+        if self.options['sourceDir'] == "":
+            for line in sys.stdin:
+                line = line.strip()
+                if os.path.isfile(line):
+                    allFiles.append(line.strip())
+        else:
+            for root, subFolders, files in os.walk(self.options['sourceDir']):
```
Comment (pieper) [inline dicomsort.py]: I think it would be cleaner if the filelist handling were factored out, so that the `allFiles` variable was populated by a method that handles the implementation.  I'm not sure it's required, but someday we might want to be able to have the filelist from stdin include directories and not just files, so the code for walking through the directory tree could be re-used to handle a directory path that come either from the command line, or from a file, or from stdin.

File: README.md (hunk 1 above)
Comment (BlakeGFitch) [reply README.md]: I agree with you! I thought through those admittedly preferred options. But, I was looking for something that wouldn't impact the code too much to review easily.

With "-" passed as the sourceDir, one would have to differentiate it from other "-x" and "--argStr" style args.

With "-l" "--fileList" style, there would be the first arg to take a parameter and it would change the expectation of how many positional args followed.

I was hoping the "" solution would fit with the current code since there are so few lines of code and that it if were ripped out later, it wouldn't be a big loss.

File: README.md (hunk 2 above)
Comment (BlakeGFitch) [reply README.md]: Yes. I agree with that. I apologize. In the interest of full disclosure, there is a typo-bug fix in a print string as well.

**Comment by pieper** [comment]
I see your points - I took another look through the code and it's still pretty simple and readable.  

It would be nice to add your little find piped to grep piped to dicomsort example to usage section of the readme file.

**Comment by BlakeGFitch** [comment]
Thank you! 

I updated the README.md with an example and created a PR. I hope it looks ok.
